include/zephyr/device.h

```diff
@@ -961,21 +1065,24 @@ __syscall int device_init(const struct device *dev);
  * @param api_ Reference to device API ops.
  * @param state_ Reference to device state.
  * @param deps_ Reference to device dependencies.
+ * @param dev_id_ Device identifier token, as passed to Z_DEVICE_BASE_DEFINE
  */
-#define Z_DEVICE_INIT(name_, pm_, data_, config_, api_, state_, deps_,                   \
-			constraints_size_, constraints_)                                 \
-	{                                                                                \
-		.name = name_,                                                           \
-		.config = (config_),                                                     \
-		.api = (api_),                                                           \
-		.state = (state_),                                                       \
-		.data = (data_),                                                         \
-		IF_ENABLED(CONFIG_DEVICE_DEPS, (.deps = (deps_),)) /**/                  \
-		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,                          \
-			(.pm_constraints = (constraints_),))                             \
-		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,                          \
-			(.pm_constraints_size = (constraints_size_),))                   \
-		IF_ENABLED(CONFIG_PM_DEVICE, ({ .pm_base = (pm_),})) /**/                \
+#define Z_DEVICE_INIT(name_, pm_, data_, config_, api_, state_, deps_,			\
+			constraints_size_, constraints_, dev_id_)			\
+	{										\
+		.name = name_,								\
+		.config = (config_),							\
+		.api = (api_),								\
+		.state = (state_),							\
+		.data = (data_),							\
+		IF_ENABLED(CONFIG_DEVICE_DEPS, (.deps = (deps_),)) /**/			\
+		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,				\
+			(.pm_constraints = (constraints_),))				\
+		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,				\
+			(.pm_constraints_size = (constraints_size_),))			\
+		IF_ENABLED(CONFIG_PM_DEVICE, ({ .pm_base = (pm_),})) /**/		\
+		IF_ENABLED(CONFIG_DEVICE_DT_METADATA,					\
```

Comment:
Interestingly, this never compiled when BOTH `CONFIG_PM_DEVICE` and `CONFIG_DEVICE_DT_METADATA` were true at the same time, because it was missing a comma between the two. Is that an unusual `CONFIG_` combination?

Fixed yesterday (among other CONFIG_DEVICE_DT_METADATA issue(s)) by @gmarull in commit 91e16f16b35c / PR #74970.

In #70403, I'm now adding both `CONFIG_PM_DEVICE` and `CONFIG_DEVICE_DT_METADATA` to `tests/lib/cpp/cxx/prj.conf`, which would have caught this. This was not a C++ specific bug but who cares - the purpose of that test is maximum coverage anyway.